### PR TITLE
Feature/df 1543  kk sorting

### PR DESCRIPTION
**Pull request by emina-str**
- cleanup a bit (mainly renaming and moving components)
- implementing the SORTING option for ChFiche, CompetenceSets, CompetenceElements

### CODE REVIEW

File: alv-portal-ui/src/app/competence-catalog/shared/constants.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 export const DEFAULT_PAGE_SIZE = 20;
 export const DEFAULT_SORT = {
```
Comment (ganqqwerty) [inline alv-portal-ui/src/app/competence-catalog/shared/constants.ts]: let's rename it to DEFAULT_SORTING_OPTIONS

File: alv-portal-ui/src/app/competence-catalog/competence-elements/competence-elements-overview/competence-elements-overview.component.html [hunk 2]
```diff
@@ -54,7 +55,7 @@ <h2 class="pr-5 mb-4 d-flex">
                                  [superTitle]="competenceElement.id"
                                  [isItemClickable]="true"
                                  [showActionButtons]="true"
-                                 [actions]="[backlinkCompetenceElementAction, deleteCompetenceElementAction]"
+                                 [actions]="actions$ | async"
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/competence-elements/competence-elements-overview/competence-elements-overview.component.html]: This is related to DF-1544. Will produce error while it is not implemented in typescript.
Please remove it from this branch.

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-delete-modal/competence-set-delete-modal.component.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<ng-container *ngIf="(chFiches$ | async) as chFiches">
+<ng-container *ngIf="(chFiches$ | async) as chFiches; else placeholder">
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-delete-modal/competence-set-delete-modal.component.html]: This is related to DF-1568.
Please remove it from this branch.

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-delete-modal/competence-set-delete-modal.component.html [hunk 4]
```diff
@@ -24,3 +24,6 @@
     </ng-template>
   </alv-modal>
 </ng-container>
+<div #placeholder>
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-delete-modal/competence-set-delete-modal.component.html]: This is related to DF-1568.
Please remove it from this branch.

File: alv-portal-ui/src/app/competence-catalog/shared/shared-competence-catalog.types.ts [hunk 5]
```diff
@@ -55,4 +55,21 @@ export enum CompetenceCatalogAction {
...
+  type: SortType;
+  icon: SortIcon;
+}
+
+export enum SortType {
+  CREATED_DATE_DESC = 'CREATED_DATE_DESC',
+  CREATED_DATE_ASC = 'CREATED_DATE_ASC',
+  ALPHA_DESC = 'ALPHA_DESC',
+  ALPHA_ASC = 'ALPHA_ASC'
+}
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/shared/shared-competence-catalog.types.ts]: Different naming on SortType, SortIcon and DEFAULT_SORT_OPTIONS

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/competence-set-search-modal/competence-set-search-modal.component.ts [hunk 6]
```diff
@@ -57,7 +57,7 @@ export class CompetenceSetSearchModalComponent implements OnInit {
     return this.competenceSetRepository.search({
       page: 0,
       size: DEFAULT_PAGE_SIZE,
-      sort: DEFAULT_SORT.asc,
+      sort: DEFAULT_SORT_OPTIONS.alpha_asc,
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/competence-set-search-modal/competence-set-search-modal.component.ts]: Not refactored

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-element-search-modal/competence-element-search-modal.component.ts [hunk 7]
```diff
@@ -68,7 +68,7 @@ export class CompetenceElementSearchModalComponent implements OnInit {
     return this.competenceElementRepository.search({
       page: 0,
       size: DEFAULT_PAGE_SIZE,
-      sort: DEFAULT_SORT.asc,
+      sort: DEFAULT_SORT_OPTIONS.alpha_asc,
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/competence-sets/competence-element-search-modal/competence-element-search-modal.component.ts]: Not refactored

File: alv-portal-ui/src/app/competence-catalog/shared/shared-competence-catalog.types.ts [hunk 8]
```diff
@@ -55,4 +55,21 @@ export enum CompetenceCatalogAction {
...
+  DATE_DESC = 'DATE_DESC',
+  DATE_ASC = 'DATE_ASC',
+  ALPHA_DESC = 'ALPHA_DESC',
+  ALPHA_ASC = 'ALPHA_ASC'
+}
 
+export enum SortIcon {
+  NUMERIC_DOWN = 'sort-numeric-down',
+  NUMERIC_UP = 'sort-numeric-up',
+  ALPHA_ASC = 'sort-alpha-down',
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/shared-competence-catalog.types.ts]: @ganqqwerty  this is wrong naming, I think. should be left as ALPHA_DOWN and ALPHA_UP as before. It's related to the name of the icon.

Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/shared/shared-competence-catalog.types.ts]: I think we don't need to be bound to the icon name. DESC and ASC is related to the purpose of the sorting whereas the icon name can always change.

**Pull request merged**